src/io/psse.jl

```diff
@@ -106,36 +106,50 @@ function create_starbus_from_transformer(pm_data::Dict, transformer::Dict)::Dict
 end
 
 
+"Imports remaining keys from `data_in` into `data_out`, excluding keys in `exclude`"
+function import_remaining!(data_out::Dict, data_in::Dict; exclude=[])
+    for (k, v) in data_in
+        if k ∉ exclude
+            data_out[lowercase(k)] = v
+        end
+    end
+end
+
+
 """
     psse2pm_branch!(pm_data, pti_data)
 
 Parses PSS(R)E-style Branch data into a PowerModels-style Dict.
 """
-function psse2pm_branch!(pm_data::Dict, pti_data::Dict)
+function psse2pm_branch!(pm_data::Dict, pti_data::Dict, import_all::Bool)
     pm_data["branch"] = []
     if haskey(pti_data, "BRANCH")
         for (i, branch) in enumerate(pti_data["BRANCH"])
             sub_data = Dict{String,Any}()
 
-            sub_data["f_bus"] = branch["I"]
-            sub_data["t_bus"] = branch["J"]
-            sub_data["br_r"] = branch["R"]
-            sub_data["br_x"] = branch["X"]
-            sub_data["g_fr"] = branch["GI"]
-            sub_data["b_fr"] = branch["BI"] == 0. && branch["B"] != 0. ? branch["B"] / 2 : branch["BI"]
-            sub_data["g_to"] = branch["GJ"]
-            sub_data["b_to"] = branch["BJ"] == 0. && branch["B"] != 0. ? branch["B"] / 2 : branch["BJ"]
-            sub_data["rate_a"] = branch["RATEA"]
-            sub_data["rate_b"] = branch["RATEB"]
-            sub_data["rate_c"] = branch["RATEC"]
+            sub_data["f_bus"] = pop!(branch, "I")
+            sub_data["t_bus"] = pop!(branch, "J")
+            sub_data["br_r"] = pop!(branch, "R")
+            sub_data["br_x"] = pop!(branch, "X")
+            sub_data["g_fr"] = pop!(branch, "GI")
+            sub_data["b_fr"] = branch["BI"] == 0. && branch["B"] != 0. ? branch["B"] / 2 : pop!(branch, "BI")
+            sub_data["g_to"] = pop!(branch, "GJ")
+            sub_data["b_to"] = branch["BJ"] == 0. && branch["B"] != 0. ? pop!(branch, "B") / 2 : pop!(branch, "BJ")
+            sub_data["rate_a"] = pop!(branch, "RATEA")
+            sub_data["rate_b"] = pop!(branch, "RATEB")
+            sub_data["rate_c"] = pop!(branch, "RATEC")
             sub_data["tap"] = 1.0
             sub_data["shift"] = 0.0
-            sub_data["br_status"] = branch["ST"]
+            sub_data["br_status"] = pop!(branch, "ST")
             sub_data["angmin"] = 0.0
             sub_data["angmax"] = 0.0
             sub_data["transformer"] = false
             sub_data["index"] = i
 
+            if import_all
+                import_remaining!(sub_data, branch)
+            end
+
             append!(pm_data["branch"], [deepcopy(sub_data)])
```

Comment:
Do we need to do a deepcopy here?  Also, I think append will still work if this is not wrapped in an array.  Same suggestion for the same block in other functions.

Reply:
deepcopy I used because everything is done in a loop, reusing the same sub_data, and i just wanted to be sure i wasn't passing a pointer, which I believe we discussed at one point early on. if you feel confident that its not needed, its no problem to take it out. `append!` does require the second item to be a collection, but in checking this i realize i could use `push!`

Reply:
aaa yes, I am most often using `push!`